python/ray/_raylet.pyx

```diff
@@ -2093,38 +2159,83 @@ cdef class CoreWorker:
                 with nogil:
                     check_status(
                         CCoreWorkerProcess.GetCoreWorker().SealReturnObject(
-                            return_id, return_ptr[0]))
+                            return_id, return_ptr[0], generator_id))
             return True
         else:
             with nogil:
                 success = (CCoreWorkerProcess.GetCoreWorker()
-                           .PinExistingReturnObject(return_id, return_ptr))
+                           .PinExistingReturnObject(
+                                   return_id, return_ptr, generator_id))
             return success
 
-    cdef store_task_outputs(
-            self, worker, outputs, const c_vector[CObjectID] return_ids,
-            c_vector[shared_ptr[CRayObject]] *returns):
+    cdef store_task_outputs(self,
+                            const CAddress &caller_address,
+                            worker, outputs,
+                            is_dynamic,
+                            const CObjectID &outer_id,
```

Comment:
Are they duplicated: if outer_id is not Nil then is_dynamic is True?

Reply:
Yeah I'll dedupe.